### ISSUE TITLE
response.body should be optional

### ISSUE THREAD

**Issue by Rich-Harris** [sveltejs/kit#1047]
**Describe the bug**
This... https://github.com/sveltejs/kit/blob/9d8dd08bc897b0951923993184c86a142e10990d/packages/kit/src/runtime/server/endpoint.js#L19-L27 ...is incorrect. Some responses (e.g. 3xx responses) shouldn't have bodies.

**To Reproduce**
Create an endpoint that returns a response without a body. Visit it.

**Expected behavior**
Body should either be optional, or only mandated for certain status codes.

**Severity**
It's easy to work around, but it's a bit odd

**Comment by RA9** [comment]
I would like to work on this issue, I would prefer the expected behavior of the `body` to be optional like that it gives flexibility but will love to hear from @Rich-Harris what those status codes are.

**Comment by Rich-Harris** [comment]
Thanks — the relevant RFC is at https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html, but it's a little hard to parse. For now it's probably easiest to just always make the body optional

**Comment by RA9** [comment]
Thanks will get to work now

**Comment by kamholz** [comment]
Related issue: I noticed that you can't just return `{ status: 200 }`, but you need to include a `body`. It would be nice to make body optional because for some API endpoint cases no body needs to be returned, but there could be other issues I'm not considering.
